wdl-lint/src/lib.rs

```diff
@@ -42,6 +42,14 @@ pub use tags::*;
 pub use visitor::*;
 pub use wdl_ast as ast;
 
+/// The reserved rule names that are used by analysis.
+const RESERVED_RULE_NAMES: &[&str] = &[
```

Comment:
This seems backwards to me. `wdl-lint` comes before `wdl-analysis` in the dependency tree, so `wdl-lint` should have "first dibs" on rule names, and `wdl-analysis` has to ensure there's no overlap.

Reply:
`wdl-analysis` doesn't depend on `wdl-lint` (I'm not sure why there's an optional dependency in `wdl-analysis`'s `Cargo.toml`, but it never gets activated; it should be removed), so they're actually independent.

`wdl-analysis` doesn't know anything about what validation rules are running; it's just given a validator to use.

I feel like the rules should be reserved inside of `wdl-lint` as analysis is a required step whereas linting is always viewed as "additional".

I can think about this more, though.

Reply:
Ah I misunderstood. That actually makes more sense now that you point it out, they are quite independent.

I'll think about this some more as well. I'm not sure what the right solution is. I think this is probably ok for at least the short-term, but we may be able to come up with something better for long term.